File: Data Structures & Algorithms/two-integer-sum/submission-5.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for n in nums:
            m = target - n
            if m in nums and nums.index(n)!=nums.index(m): 
                return [nums.index(n), nums.index(m)]
            elif nums.index(n)==nums.index(m): 
                i = nums.index(n)
                nums.remove(n)
                if m in nums:
                    return [i, nums.index(m)]
            else: 
                pass

File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        int i = 0;
        while (n>i) { 
            int sum = numbers[n-1] + numbers[i];
            if (sum > target) {
                n--;
            }
            else if (sum < target) {
                i++;
            }
        }

        return {i+1, n};
    }
};
